transport/src/test/java/io/scalecube/transport/TransportTest.java

```diff
@@ -355,20 +353,11 @@ public void testCompleteObserver() throws Exception {
     final CompletableFuture<Boolean> completeLatch = new CompletableFuture<>();
     final CompletableFuture<Message> messageLatch = new CompletableFuture<>();
 
-    server.listen().subscribe(new Subscriber<Message>() {
-      @Override
-      public void onCompleted() {
-        completeLatch.complete(true);
-      }
-
-      @Override
-      public void onError(Throwable e) {}
-
-      @Override
-      public void onNext(Message message) {
-        messageLatch.complete(message);
-      }
-    });
+    server.listen().subscribe(messageLatch::complete,
```

Comment:
Need some clarification such complex mechanism

Reply:
can you explain what you refer as complex?
why use Latch you mean?

Reply:
I mean that potentially we may avoid the `CompletableFuture` usage and replace that with `Mono`. However, that refactoring needs additional investigation sitting together with someone who is aware in the logic. Actually, we may do a call and discuss that later and fix in the separate PR